src/PointwiseFunctions/AnalyticData/SelfForce/GeneralRelativity/CircularOrbit.cpp

```diff
@@ -0,0 +1,286 @@
+                             const int m_mode_number)
+    : black_hole_mass_(black_hole_mass),
+      black_hole_spin_(black_hole_spin),
+      orbital_radius_(orbital_radius),
+      m_mode_number_(m_mode_number) {}
+
+CircularOrbit::CircularOrbit(CkMigrateMessage* m)
+    : elliptic::analytic_data::Background(m),
+      elliptic::analytic_data::InitialGuess(m) {}
+
+tnsr::I<double, 2> CircularOrbit::puncture_position() const {
+  const double M = black_hole_mass_;
+  const double r_plus = M * (1. + sqrt(1. - square(black_hole_spin_)));
+  const double r_0 = orbital_radius_;
+  const double r_star = gr::tortoise_radius_from_boyer_lindquist_minus_r_plus(
+      r_0 - r_plus, M, black_hole_spin_);
+  return tnsr::I<double, 2>{{{r_star, M_PI_2}}};
+}
+
+// Background
+tuples::TaggedTuple<Tags::Alpha, Tags::Beta, Tags::GammaRstar, Tags::GammaTheta>
+CircularOrbit::variables(const tnsr::I<DataVector, 2>& x,
+                         tmpl::list<Tags::Alpha, Tags::Beta, Tags::GammaRstar,
+                                    Tags::GammaTheta> /*meta*/) const {
+  const double a = black_hole_spin_ * black_hole_mass_;
+  const double M = black_hole_mass_;
+  const double r_plus = M * (1. + sqrt(1. - square(black_hole_spin_)));
+  const double r_minus = M * (1. - sqrt(1. - square(black_hole_spin_)));
+  const double r_0 = orbital_radius_;
+  const double omega = 1. / (a + sqrt(cube(r_0) / M));
+  const auto& r_star = get<0>(x);
+  const auto& theta = get<1>(x);
+  const DataVector cos_theta = cos(theta);
+  const DataVector r_minus_r_plus =
+      gr::boyer_lindquist_radius_minus_r_plus_from_tortoise(r_star, M,
+                                                            black_hole_spin_);
+  const DataVector r = r_minus_r_plus + r_plus;
+  const DataVector delta = r_minus_r_plus * (r - r_minus);
+  const DataVector r_sq_plus_a_sq = square(r) + square(a);
+  const DataVector r_sq_plus_a_sq_sq = square(r_sq_plus_a_sq);
+  const DataVector sin_theta_squared = 1. - square(cos_theta);
+  const DataVector sigma_squared =
+      r_sq_plus_a_sq_sq - square(a) * delta * sin_theta_squared;
+  tuples::TaggedTuple<Tags::Alpha, Tags::Beta, Tags::GammaRstar,
+                      Tags::GammaTheta>
+      result{};
+  auto& alpha = get<Tags::Alpha>(result);
+  auto& beta = get<Tags::Beta>(result);
+  auto& gamma_rstar = get<Tags::GammaRstar>(result);
+  auto& gamma_theta = get<Tags::GammaTheta>(result);
+  const size_t num_points = r.size();
+  get(alpha) = delta / r_sq_plus_a_sq_sq;
+  for (size_t i = 0; i < beta.size(); ++i) {
+    beta[i] = ComplexDataVector{num_points, 0.};
+    gamma_rstar[i] = ComplexDataVector{num_points, 0.};
+    gamma_theta[i] = ComplexDataVector{num_points, 0.};
+  }
+  const ComplexDataVector temp1 =
+      1. / r * std::complex<double>(0., 2. * a * m_mode_number_);
+  // tt, tr, ttheta, tphi, rr, rtheta, rphi, theta theta, theta phi, phi phi
+  double Areal[10][10], Aimag[10][10], Breal[10][10], Bimag[10][10],
```

Comment:
Can you do `std::array<std::array<double, 10>, 10>`, `Areal.data()`, and then `gsl::at` below?